src/o365/spo/commands/web/web-add.ts

```diff
@@ -0,0 +1,313 @@
+      this.getRequestDigestForSite(subsiteFullUrl, siteAccessToken, cmd, debug)
+      .then((res: ContextInfo): Promise<any> => {
+        let requestOptions: any = {
+          url: `${subsiteFullUrl}/_vti_bin/client.svc/ProcessQuery`,
+          headers: Utils.getRequestHeaders({
+            authorization: `Bearer ${siteAccessToken}`,
+            'X-RequestDigest': res.FormDigestValue
+          }),
+          body: `<Request xmlns="http://schemas.microsoft.com/sharepoint/clientquery/2009" SchemaVersion="15.0.0.0" LibraryVersion="16.0.0.0" ApplicationName="Javascript Library"><Actions><ObjectPath Id="1" ObjectPathId="0" /><ObjectPath Id="3" ObjectPathId="2" /><ObjectPath Id="5" ObjectPathId="4" /><SetProperty Id="6" ObjectPathId="4" Name="UseShared"><Parameter Type="Boolean">true</Parameter></SetProperty><Query Id="7" ObjectPathId="4"><Query SelectAllProperties="true"><Properties><Property Name="UseShared" ScalarProperty="true" /></Properties></Query></Query></Actions><ObjectPaths><StaticProperty Id="0" TypeId="{3747adcd-a3c3-41b9-bfab-4a64dd2f1e0a}" Name="Current" /><Property Id="2" ParentId="0" Name="Web" /><Property Id="4" ParentId="2" Name="Navigation" /></ObjectPaths></Request>`
+        };
+
+        return request.post(requestOptions).then((res:any) : any => {
+          if (debug) {
+            cmd.log(`Response : SetInheritNavigation`);
+            cmd.log(res);
+          }
+          return resolve(true);
+        }, (err: any) =>  { 
+          cmd.log(new CommandError(`Failed to set inheritNavigation for the web - ${subsiteFullUrl}`)); 
+          return reject(err);
+        });
+      });
+    });
+  }
+
+  public get name(): string {
+    return commands.WEB_ADD;
+  }
+
+  public get description(): string {
+    return 'Creates new subsite';
+  }
+
+  protected requiresTenantAdmin(): boolean {
+    return false;
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.title = (!(!args.options.title)).toString();
+    telemetryProps.webUrl = (!(!args.options.webUrl)).toString();
+    telemetryProps.webTemplate = (!(!args.options.webTemplate)).toString();
+    telemetryProps.parentWebUrl = (!(!args.options.parentWebUrl)).toString();
+    telemetryProps.description = (!(!args.options.description)).toString();
+    telemetryProps.locale = (!(!args.options.locale)).toString();
+    telemetryProps.breakInheritance = args.options.breakInheritance || false;
+    telemetryProps.inheritNavigation = args.options.inheritNavigation || false;
+
+    return telemetryProps;
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: () => void): void {
+    const resource: string = Auth.getResourceFromUrl(args.options.parentWebUrl);
+    let siteAccessToken: string = '';
+    
+    auth
+    .ensureAccessToken(resource, cmd, this.debug)
```

Comment:
`ensureAccessToken` gets the token for the currently connected site rather than the specified site. Additionally, it alters the current context. Instead, you should use `getAccessToken`. This is explained in more detail at https://sharepoint.github.io/office365-cli/concepts/authorization-tokens/